### PR TITLE
Exposes ray init functionality

### PR DESCRIPTION
**Pull request by skrawcz**
This was bothering me that we hadn't exposed a few things, this fixes that. It means that someone can now connect to a ray cluster for example.

## Changes
 - h_ray.py task executor

## How I tested this
 - locally

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/plugins/h_ray.py [hunk 1]
```diff
@@ -209,20 +209,33 @@ class RayTaskExecutor(executors.TaskExecutor):
     This is still experimental, so the API might change.
     """
 
-    def __init__(self, num_cpus: int):
+    def __init__(
+        self,
+        num_cpus: int = None,
```
Comment (elijahbenizzy) [inline hamilton/plugins/h_ray.py:214]: We expose num_cpus -- anything else we want to expose? Or do we just want to keep that there for backwards compatibility?

File: hamilton/plugins/h_ray.py [hunk 2]
```diff
@@ -209,20 +209,33 @@ class RayTaskExecutor(executors.TaskExecutor):
...
+        config: typing.Dict[str, typing.Any] = None,
```
Comment (elijahbenizzy) [inline hamilton/plugins/h_ray.py:215]: `ray_init_config` might be more descriptive

**Review by elijahbenizzy — COMMENTED** [review]
A few nits but let's ship.

File: hamilton/plugins/h_ray.py (hunk 1 above)
Comment (skrawcz) [reply hamilton/plugins/h_ray.py:214]: backwards compatibility.

File: hamilton/plugins/h_ray.py (hunk 2 above)
Comment (skrawcz) [reply hamilton/plugins/h_ray.py]: done

**Review by elijahbenizzy — APPROVED** [review]

**Pull request merged**